### DISCUSSION TITLE
UI feedback in annotation query editor

### DISCUSSION THREAD

**Discussion by gtk-grafana** [grafana/grafana#53974] [Observability]
Currently in TSDB, prometheus, and graphite the annotation editors don't return error messages from the datasources, but some datasources (such as flux) do.

Prometheus: no feedback at all if query returns error.
<img width="1505" alt="image" src="https://user-images.githubusercontent.com/109082771/185667945-1bb0f4ab-3cd0-4b16-9578-f68c5b0f7989.png">

Graphite at least returns "no results found" message, and TDSB will follow suit in 9.2 with [#53856](https://github.com/grafana/grafana/pull/53856)
<img width="1524" alt="image" src="https://user-images.githubusercontent.com/109082771/185668149-29f4e918-c3e5-4774-b642-6672b3f9756f.png">

Flux provides the API error visible to the user.
<img width="1533" alt="image" src="https://user-images.githubusercontent.com/109082771/185670665-28b9739e-6e3c-42de-bd01-bd5129af9bbb.png">

Suggestion: we update datasource annotation editors to provide UI feedback that a query was successful, or include relevant error info. Makes debugging annotations much easier!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
